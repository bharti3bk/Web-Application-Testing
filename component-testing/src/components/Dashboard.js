import React, { useState } from "react";
import Display from "./Display";

export default function Dashboard(){
    const[ball ,setballs] = useState(0);
    const[strike ,setStrike] =useState(0);
}
