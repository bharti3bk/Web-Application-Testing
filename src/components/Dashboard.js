import React, { useState } from "react";
import Display from "./Display";
import { Select } from "semantic-ui-react";

const activities = [
  { key: "st", value: "st", text: "Strike" },
  { key: "ba", value: "ba", text: "Ball" },
  { key: "hi", value: "hi", text: "Hit" },
  { key: "fo", value: "fo", text: "Foul" }
];

export default function Dashboard({intitialActivity = "", initialStrikes = 0, initialBalls = 0}) {
  const [state, setState] = useState({ balls: initialBalls, strikes: initialStrikes, activity:intitialActivity });

  function updateStats() {
    switch(state.activity) {
        case "st":
            if(state.strikes === 2) 
                setState({...state,balls: 0, strikes: 0})
            else 
                setState({...state, strikes: state.strikes + 1})
            break;
        case "ba":
            if(state.balls === 3) 
                setState({...state,balls: 0, strikes: 0})
             else 
                setState({...state,balls: state.balls + 1})
            break;
        case "hi":
            setState({...state,balls: 0 , strikes: 0})
            break;
        case "fo":
            if(state.strikes < 2) 
                setState({...state,strikes: state.strikes + 1})
            break;
        default:
            console.log("activity is not selected")
    }
  }
  return (
    <div>
        <Select placeholder="Select Activity" options={activities} onChange={(e, data) => state.activity = data.value } />
        <button onClick={() => updateStats()}> Update Stats</button>
        <Display balls={state.balls} strikes={state.strikes} />
    </div>
  );
}
