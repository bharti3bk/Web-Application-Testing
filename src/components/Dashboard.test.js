import React from "react";
import {cleanup, fireEvent, render} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Dashboard from "./Dashboard";

afterEach(cleanup)

it('renders ', () => {
    const {asFragment} = render(<Dashboard />)
    expect(asFragment()).toMatchSnapshot()
});

it('initializes with 0 ball and 0 strikes', () => {
    const {getByText} = render(<Dashboard />)

    expect(getByText("Balls: 0")).toHaveTextContent("0")
    expect(getByText("Strikes: 0")).toHaveTextContent("0")
})

it('updateStats sets the strikes to 1 when selected option is strike and update stats button is clicked', () => {
    // Arrange
    const {getByText} = render(<Dashboard intitialActivity="st" initialStrikes={0} />)

    // Act
    fireEvent.click(getByText("Update Stats"))

    // Assert
    expect(getByText("Strikes: 1")).toHaveTextContent("1")
})

it('updateStats sets the strike to 0 when stikes are already 2 and another strike is made when update stats button is clicked', () => {
    // Arrange
    const {getByText} = render(<Dashboard initialStrikes={2} intitialActivity="st" /> )

    // Act
    fireEvent.click(getByText("Update Stats"))

    // Assert
    expect(getByText("Strikes: 0")).toHaveTextContent("0")
})