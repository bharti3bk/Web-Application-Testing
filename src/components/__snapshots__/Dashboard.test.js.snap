// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders  1`] = `
<DocumentFragment>
  <div>
    <div
      aria-expanded="false"
      class="ui selection dropdown"
      role="listbox"
      tabindex="0"
    >
      <div
        aria-atomic="true"
        aria-live="polite"
        class="default text"
        role="alert"
      >
        Select Activity
      </div>
      <i
        aria-hidden="true"
        class="dropdown icon"
      />
      <div
        class="menu transition"
      >
        <div
          aria-checked="false"
          aria-selected="true"
          class="selected item"
          role="option"
          style="pointer-events: all;"
        >
          <span
            class="text"
          >
            Strike
          </span>
        </div>
        <div
          aria-checked="false"
          aria-selected="false"
          class="item"
          role="option"
          style="pointer-events: all;"
        >
          <span
            class="text"
          >
            Ball
          </span>
        </div>
        <div
          aria-checked="false"
          aria-selected="false"
          class="item"
          role="option"
          style="pointer-events: all;"
        >
          <span
            class="text"
          >
            Hit
          </span>
        </div>
        <div
          aria-checked="false"
          aria-selected="false"
          class="item"
          role="option"
          style="pointer-events: all;"
        >
          <span
            class="text"
          >
            Foul
          </span>
        </div>
      </div>
    </div>
    <button>
       Update Stats
    </button>
    <div
      class="display"
    >
      <p>
         Balls: 0
      </p>
      <p>
         Strikes: 0
      </p>
    </div>
  </div>
</DocumentFragment>
`;
